ntoskrnl/ke/i386/kiinit.c

```diff
@@ -159,7 +159,25 @@ KiInitMachineDependent(VOID)
             CpuCount = KeNumberProcessors;
             KeIpiGenericCall(Ki386EnableXMMIExceptions, (ULONG_PTR)&CpuCount);
 
-            /* FIXME: Implement and enable XMM Page Zeroing for Mm */
+            /* Patch the KeZeroPages function to enable fast zeroing */
+            PUCHAR Instr = (PUCHAR)KeZeroPages;
```

Comment:
Isn't it possible to patch the symbol itself?

Reply:
> Isn't it possible to patch the symbol itself?

This function is called from inside the kernel. Patching the symbol table (is that waht you mean ?) wouldn't help

Reply:
> I'm guessing a function pointer wouldn't incur any noticeable perf cost given we always zero an entire page.

Yes, but then you add complexity to the code for little benefit. I guess it's a matter of taste :-)

Reply:
You think a function pointer is more complex than runtime patching assembly? Interesting take! I guess it is a matter of taste then :)

Reply:
Please use a function pointer. There no additional complexity. You could also declare the function as dllimport and then have a pointer `PVOID __imp_KeZeroPages;`, which is the same thing.